tests/bluetooth/bsim/mesh/src/test_dfu.c

```diff
@@ -185,10 +248,23 @@ static void target_dfu_transfer_end(struct bt_mesh_dfu_srv *srv, const struct bt
 	ASSERT_TRUE(expect_dfu_xfer_end);
 	ASSERT_TRUE(success);
 
+	if (expected_stop_phase == DFU_SRV_STOP_VERIFY) {
+		k_sem_give(&dfu_ended);
```

Comment:
General comment. Would it be possible to use `dfu_ended` semaphore only in `target_dfu_apply` (on the target node) and in `dfu_cli_ended` (on the client) and use separate semaphores for each dfu step? What I don't really like now is that it becomes a test inside the test and supporting this becomes a problem. Perhaps we could do the other way around, but then the other tests need to be updated.